worlds/lingo/__init__.py

```diff
@@ -0,0 +1,149 @@
+from .items import LingoItem, StaticLingoItems
+from .locations import LingoLocation, StaticLingoLocations
+from .Options import lingo_options, get_option_value
+from .testing import LingoTestOptions
+from worlds.generic.Rules import set_rule
+from .rules import LingoLogic, set_rules
+from .player_logic import LingoPlayerLogic
+from .regions import create_regions
+from math import floor
+
+
+class LingoWebWorld(WebWorld):
+    theme = "grass"
+    tutorials = [Tutorial(
+        "Multiworld Setup Guide",
+        "A guide to playing Lingo with Archipelago.",
+        "English",
+        "setup_en.md",
+        "setup/en",
+        ["hatkirby"]
+    )]
+
+
+class LingoWorld(World):
+    """
+    Lingo is a first person indie puzzle game in the vein of The Witness. You find yourself in a mazelike, non-Euclidean
+    world filled with 800 word puzzles that use a variety of different mechanics.
+    """
+    game = "Lingo"
+    web = LingoWebWorld()
+
+    base_id = 444400
+    topology_present = True
+    data_version = 1
+
+    static_logic = StaticLingoLogic()
+    static_items = StaticLingoItems(base_id)
+    static_locat = StaticLingoLocations(base_id)
+    option_definitions = lingo_options
+
+    item_name_to_id = {
+        name: data.code for name, data in static_items.ALL_ITEM_TABLE.items()
+    }
+    location_name_to_id = {
+        name: data.code for name, data in static_locat.ALL_LOCATION_TABLE.items()
+        if data.code is not None
+    }
+
+    # This is just used for unit testing. It should remain at the default values for actual play.
+    test_options: LingoTestOptions = LingoTestOptions()
+
+    def _get_slot_data(self):
+        return {
+            'door_ids_by_item_id': {
+                data.code: data.door_ids for data in self.static_items.ALL_ITEM_TABLE.values()
+                if data.code is not None and len(data.door_ids) > 0
+            },
+            'painting_ids_by_item_id': {
+                data.code: data.painting_ids for data in self.static_items.ALL_ITEM_TABLE.values()
+                if data.code is not None and len(data.painting_ids) > 0
+            },
+            'panel_ids_by_location_id': {
+                data.code: data.panel_ids() for data in self.static_locat.ALL_LOCATION_TABLE.values()
+                if data.code is not None
+            },
+            'seed': self.multiworld.per_slot_randoms[self.player].randint(0, 1000000),
+        }
+
+    def generate_early(self):
+        self.player_logic = LingoPlayerLogic(self.multiworld, self.player, self.static_logic, self.test_options)
+
+    def create_regions(self):
+        create_regions(self.multiworld, self.player, self.static_logic, self.player_logic)
+
+    def create_items(self):
+        pool = []
+
+        for name in self.player_logic.REAL_ITEMS:
+            new_item = self.create_item(name)
+            pool.append(new_item)
+
+        for location, item in self.player_logic.EVENT_LOC_TO_ITEM.items():
```

Comment:
This seems like it could be done when you create these locations which would be faster and cleaner.